tools/preflight/exec/exec_to_pod.go

```diff
@@ -0,0 +1,104 @@
+package exec
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/url"
+
+	corev1 "k8s.io/api/core/v1"
+	client "k8s.io/client-go/kubernetes"
+	clientGoScheme "k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	tty             = false
+	podResource     = "pods"
+	execSubresource = "exec"
+	POSTMethod      = "POST"
+)
+
+// Response contain's command execution response
+type Response struct {
+	Stdout, Stderr string
+	Err            error
+}
+
+// Options passed to ExecWithOptions
+type Options struct {
+	Command []string
+
+	Namespace     string
+	PodName       string
+	ContainerName string
+
+	Executor  RemoteExecutor
+	Config    *restclient.Config
+	ClientSet *client.Clientset
+}
+
+// RemoteExecutor defines the interface accepted by the Exec command - provided for test stubbing
+type RemoteExecutor interface {
+	execute(method string, url *url.URL, config *restclient.Config, stdout, stderr io.Writer, tty bool) error
+}
+
+// DefaultRemoteExecutor is the standard implementation of remote command execution
+type DefaultRemoteExecutor struct{}
+
+func (*DefaultRemoteExecutor) execute(method string, reqURL *url.URL, config *restclient.Config, stdout, stderr io.Writer, tty bool) error {
+	exec, err := remotecommand.NewSPDYExecutor(config, method, reqURL)
+	if err != nil {
+		return err
+	}
+
+	return exec.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    tty,
+	})
+}
+
+// ExecInContainer will execute given command's in specified container and return execution response in channel `resp`.
+func (op *Options) ExecInContainer(resp chan<- *Response) {
```

Comment:
Actually it is really not necessary to use channel, wait for the goroutine to finish and get the output from channel.
Just followed the implementation approach as we have in k8s-triliovault repo.

Simple function call would also work. In this case, the program would be stalled until the exec command finishes. If we ever have to do a time consuming exec then simple function call might be the concern.

@msurudwad please let know your comments on this.

Reply:
> why channel is needed? without chhanel it can also work right?

Actually it is really not necessary to use channel, wait for the goroutine to finish and get the output from channel.
Just followed the implementation approach as we have in k8s-triliovault repo.

Simple function call would also work. In this case, the program would be stalled until the exec command finishes. If we ever have to do a time consuming exec then simple function call might be the concern.

@msurudwad please let know your comments on this.

Reply:
Channel is removed now.